cilium/cmd/config.go

```diff
@@ -73,16 +75,43 @@ func configDaemon(cmd *cobra.Command, opts []string) {
 			if err := command.PrintOutput(cfgStatus.Realized.Options); err != nil {
 				os.Exit(1)
 			}
+			if err := command.PrintOutput(cfgStatus.DaemonConfigurationMap); err != nil {
+				Fatalf("Error while printing configmap: %v", err)
+			}
 			return
 		}
-		dumpConfig(cfgStatus.Immutable)
-		dumpConfig(cfgStatus.Realized.Options)
-		fmt.Printf("%-24s %s\n", "k8s-configuration", cfgStatus.K8sConfiguration)
-		fmt.Printf("%-24s %s\n", "k8s-endpoint", cfgStatus.K8sEndpoint)
-		fmt.Printf("%-24s %s\n", "PolicyEnforcement", cfgStatus.Realized.PolicyEnforcement)
+		fmt.Print("### Read-Only Configurations ###\n\n")
+		keys := make([]string, 0, len(cfgStatus.DaemonConfigurationMap))
+		for k := range cfgStatus.DaemonConfigurationMap {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		for _, k := range keys {
+			v := cfgStatus.DaemonConfigurationMap[k]
+			if reflect.ValueOf(v).Kind() == reflect.Map {
+				mapString := make(map[string]string)
+				m, ok := v.(map[string]interface{})
+				if ok {
+					for key, value := range m {
+						mapString[key] = fmt.Sprintf("%v", value)
+					}
+					dumpConfig(mapString)
+					continue
+				} else {
+					fmt.Fprintf(os.Stderr, "Error: cannot cast daemon config map to map[string]interface{}\n")
+				}
+			}
+			fmt.Printf("%-34s: %v\n", k, v)
```

Comment:
If `ok` is `false` above then this line will run. That doesn't seem right. I would recommend putting this in an `else` statement.

Reply:
Sorry for the delay. Thanks for pointing that out. I will put a fix.